### PR TITLE
Module hangs if it imports an errored module

### PR DESCRIPTION
**Pull request by AtOMiCNebula**
ResolveExternalModuleDependencies has no problem today resolving against a module that has already been determined to be in an error state.  In this case, it has already notified its dependents at the time of its failure, and no further notification will occur.  As a result, the module hangs if we do not handle this scenario.

This fix addresses the issue by not proceeding if the module being imported is already in an error state.  The error is passed upward.

### CODE REVIEW

**Comment by AtOMiCNebula** [comment]
@boingoing @akroshg Baby's first ChakraCore change! ;)

**Comment by akroshg** [comment]
Could you create a native unittest for this?
this PR  #1803 has added native unittests to craft such scenario.

**Comment by Yongqu** [comment]
LGTM

File: lib/Runtime/Language/SourceTextModuleRecord.cpp [hunk 1]
```diff
@@ -587,8 +594,12 @@ namespace Js
             });
             if (FAILED(hr))
             {
-                JavascriptError *error = scriptContext->GetLibrary()->CreateError();
-                JavascriptError::SetErrorMessageProperties(error, hr, _u("fetch import module failed"), scriptContext);
+                if (error == nullptr)
+                {
```
Comment (AtOMiCNebula) [inline lib/Runtime/Language/SourceTextModuleRecord.cpp]: I want to propagate the inner error outward, so I only create the "fetch import module failed" error if we don't have a better one to use.

**Comment by dilijev** [comment]
@boingoing @AtOMiCNebula Is this PR waiting on anything? Can we get that test added and merge this in? (At this point this should probably go into release/1.4 so the PR will need to be retargeted.)

**Comment by AtOMiCNebula** [comment]
I let this one get super stale, sorry!  I hope to be able to revisit this in the not too distant future; I'll close this PR for now, and open a new PR after rebasing the change onto a more recent master. :)

**Comment by dilijev** [comment]
Thanks @AtOMiCNebula! Looking forward to the update!

**Pull request closed without merging**